addons/sale_timesheet/report/project_profitability_report_analysis.py

```diff
@@ -177,14 +177,11 @@ def init(self):
                                 FROM project_project P
                                     LEFT JOIN account_analytic_account AA ON P.analytic_account_id = AA.id
                                     LEFT JOIN account_analytic_line AAL ON AAL.account_id = AA.id
-                                WHERE AAL.amount > 0.0 AND AAL.project_id IS NULL AND P.active = 't' AND P.allow_timesheets = 't'
-                                    AND NOT EXISTS (
-                                        SELECT SOL.id
-                                        FROM sale_order_line SOL
-                                        JOIN sale_order_line_invoice_rel SOINV ON SOINV.order_line_id = SOL.id AND SOINV.invoice_line_id = AAL.move_id -- AAL.move_id is an account.move.line id
-                                        WHERE SOL.qty_delivered_method IN ('timesheet', 'manual')
-                                            OR (SOL.qty_delivered_method = 'analytic' AND SOL.invoice_status != 'no')
-                                    )
+                                    LEFT JOIN product_product PP ON PP.id = AAL.product_id
+                                    LEFT JOIN product_template PT ON PT.id = PP.product_tmpl_id
+                                    LEFT JOIN sale_order SO ON SO.analytic_account_id = AA.id
+                                    LEFT JOIN sale_order_line SOL ON SOL.product_id = AAL.product_id AND SOL.order_id = SO.id
+                                WHERE AAL.amount > 0.0 AND AAL.project_id IS NULL AND P.active = 't' AND P.allow_timesheets = 't' AND PT.service_type = 'manual' AND PT.service_tracking = 'no' AND SOL.is_expense IS NOT TRUE AND SOL.is_downpayment IS NOT TRUE
```

Comment:
I do think this will throw out the bug fix introduced [here](https://github.com/odoo/odoo/commit/4d9305d122f79792db780bbd97ae6732bf06ff9f).

May I ask you to test the case defined in this commit ?

Also, I think the clauses may be clarified and match with the definition of `qty_delivered_method = 'manual' AND SOL.is_downpayment = 'f'`. Only the clause PT.service_tracking = 'no' is more specific. I think, for the product `qty_delivered_method = 'manual'` we should not make an heuristic on counting the task logs, etc. So I would remove the `qty_delivered_method = 'manual'` [here](https://github.com/odoo/odoo/pull/73506/commits/7a5310cd9e0870e46e41f76d376ca301b9c29b29#diff-a19fc8527494e0bf2755c2ad17ba28f0da1c3d973a5c02790aee7ac724fbaaa0R284-R288) and change the clause here so that it looks like :

```suggestion
WHERE AAL.amount > 0.0 AND AAL.project_id IS NULL AND P.active = 't' AND P.allow_timesheets = 't'
                                    AND NOT EXISTS (
                                        SELECT SOL.id
                                        FROM sale_order_line SOL
                                        JOIN sale_order_line_invoice_rel SOINV ON SOINV.order_line_id = SOL.id AND SOINV.invoice_line_id = AAL.move_id -- AAL.move_id is an account.move.line id
                                        WHERE SOL.qty_delivered_method = 'timesheet'
                                            OR SOL.is_downpayment = 't' -- OR (SOL.is_downpayment = 't' AND SOL.qty_delivered_method = 'manual')
                                            OR (SOL.qty_delivered_method = 'analytic' AND SOL.invoice_status != 'no')
                                    )
```

Naively, I didn't think to the clause `SOL.is_downpayment = 't'` but I think you're right by using it !

Reply:
Okay @proose  , as `AND PT.service_type = 'manual' AND PT.service_tracking = 'no'` are the default value, the case of the preceding issue is covered, which was my fear :) :+1: 

Maybe we can add comments in the code for those two clauses so that we don't have to grep in the code base to see in which case we are.

Thanks for having convinced me ! :)